compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2362,6 +2362,19 @@ impl<'tcx> Ty<'tcx> {
     /// to represent the closure kind, because it has not yet been
     /// inferred. Once upvar inference (in `rustc_hir_analysis/src/check/upvar.rs`)
     /// is complete, that type variable will be unified.
+    ///
+    /// To be noted that you can use [`ClosureArgs::kind()`] or [`CoroutineClosureArgs::kind()`]
+    /// to get the same information, which you can get by calling [`GenericArgs::as_closure()`].
```

Comment:
or as_coroutine_closure -- depends on the type

Reply:
Adding it. :+1: